cmd/build.go

```diff
@@ -154,7 +158,69 @@ func picardImage() (string, error) {
 	return fmt.Sprintf("%s@%s", picardRepo, sha), nil
 }
 
+func configVersion(configBytes []byte) (string, bool) {
+	var raw map[string]interface{}
+	if err := yaml.Unmarshal(configBytes, &raw); err != nil {
+		return "", false
+	}
+
+	var configWithVersion struct {
+		Version string `yaml:"version"`
+	}
+	if err := mapstructure.WeakDecode(raw, &configWithVersion); err != nil {
+		return "", false
+	}
+	return configWithVersion.Version, true
+}
+
+func validateConfigVersionHelper(configFilename string) error {
+	invalidConfigError := "\nYou attempted to run a local build with version '%s' of configuration.\nLocal builds do not support that version at this time.\nYou can use 'circleci config process' to pre-process your config into a version that local builds can run (see 'circleci help config process' for more information)"
+
+	configBytes, err := ioutil.ReadFile(configFilename)
+	if err != nil {
+		return errors.Wrapf(err, "Unable to read config file")
+	}
+
+	version, isVersion := configVersion(configBytes)
+	if !isVersion || version == "" {
+		return errors.New("Unable to identify config version")
+	}
+
+	if version != "2.0" && version != "2" {
+		return fmt.Errorf(invalidConfigError, version)
+	}
+
+	return nil
+}
+
+func containsIdx(list []string, v1 string, v2 string) int {
+	for i, v := range list {
+		if v == v1 || v == v2 {
+			return i
+		}
+	}
+	return -1
+}
+
+func validateConfigVersion(args []string) error {
+	indexConfigFlag := containsIdx(args, "--config", "--c")
+	configPath := opts.configFilename
```

Comment:
I believe this line is not doing anything because  `DisableFlagParsing` is enabled.

Reply:
🤦‍♀️